Subject: i386: remove pit_interrupt_hook

From: Chris Wright <chrisw@sous-sol.org>

Remove pit_interrupt_hook as it adds just an extra layer.

Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Ingo Molnar <mingo@elte.hu>
---

 include/asm-i386/i8253.h                 |   11 -----------
 include/asm-i386/mach-default/do_timer.h |    2 +-
 include/asm-i386/mach-voyager/do_timer.h |    2 +-
 3 files changed, 2 insertions(+), 13 deletions(-)

Index: linux-2.6.22/include/asm-i386/i8253.h
===================================================================
--- linux-2.6.22.orig/include/asm-i386/i8253.h	2007-07-24 08:56:54.000000000 +0200
+++ linux-2.6.22/include/asm-i386/i8253.h	2007-07-24 08:56:54.000000000 +0200
@@ -12,17 +12,6 @@ extern spinlock_t i8253_lock;
 
 extern struct clock_event_device *global_clock_event;
 
-/**
- * pit_interrupt_hook - hook into timer tick
- * @regs:	standard registers from interrupt
- *
- * Call the global clock event handler.
- **/
-static inline void pit_interrupt_hook(void)
-{
-	global_clock_event->event_handler(global_clock_event);
-}
-
 extern void setup_pit_timer(void);
 
 #endif	/* __ASM_I8253_H__ */
Index: linux-2.6.22/include/asm-i386/mach-default/do_timer.h
===================================================================
--- linux-2.6.22.orig/include/asm-i386/mach-default/do_timer.h	2007-07-24 08:56:46.000000000 +0200
+++ linux-2.6.22/include/asm-i386/mach-default/do_timer.h	2007-07-24 08:56:54.000000000 +0200
@@ -12,5 +12,5 @@
 
 static inline void do_timer_interrupt_hook(void)
 {
-	pit_interrupt_hook();
+	global_clock_event->event_handler(global_clock_event);
 }
Index: linux-2.6.22/include/asm-i386/mach-voyager/do_timer.h
===================================================================
--- linux-2.6.22.orig/include/asm-i386/mach-voyager/do_timer.h	2007-07-24 08:56:46.000000000 +0200
+++ linux-2.6.22/include/asm-i386/mach-voyager/do_timer.h	2007-07-24 08:56:54.000000000 +0200
@@ -12,7 +12,7 @@
  **/
 static inline void do_timer_interrupt_hook(void)
 {
-	pit_interrupt_hook();
+	global_clock_event->event_handler(global_clock_event);
 	voyager_timer_interrupt();
 }
 
